src/raftstore/store/peer.rs

```diff
@@ -1763,8 +1763,9 @@ impl Peer {
     }
 
     fn handle_read(&mut self, req: RaftCmdRequest, check_epoch: bool) -> ReadResponse {
-        let mut resp = ReadExecutor::new(self.region(), &self.engines.kv, &self.tag)
-            .execute(&req, check_epoch)
+        let mut resp = ReadExecutor::new(self.engines.kv.clone(), check_epoch)
```

Comment:
Is `clone` necessary? This may cost as much as a lease check.

Reply:
We need a clone to take a snapshot.

Reply:
`raftstore::store::Snapshot` can hold an `AsRef<DB>` to resolve this. Maybe we can do it in another PR.